reports.go

```diff
@@ -46,15 +70,711 @@ type Report struct {
+	// since ID is empty we don't have anything to return. the caller should make sure if ssnTraceID is nil
+	if ssnTraceID == "" {
+		return nil, errEmptyID
+	}
+
+	rel, err := url.Parse(ssnTracePath)
+	if err != nil {
+		return nil, err
+	}
+
+	u := *c.BaseURL
+	u.Path = path.Join(u.Path, rel.String(), ssnTraceID)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.SetBasicAuth(c.APIKey, "")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, NewResponseError([]int{
+			resp.StatusCode,
+		}, resp)
+	}
+	defer func() {
+		if resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	getResp := &SSNTrace{}
+	err = json.Unmarshal(b, getResp)
+	if err != nil {
+		return nil, err
+	}
+
+	return getResp, nil
+}
+
+const sexOffenderSearchPath = "/v1/sex_offender_searches"
+
+// GetSexOffenderSearch ...
+func (r *Report) GetSexOffenderSearch(sexOffenderSearchID string, c *Client) (*SexOffenderSearch, error) {
+
+	if sexOffenderSearchID == "" {
+		return nil, errEmptyID
+	}
+
+	rel, err := url.Parse(sexOffenderSearchPath)
+	if err != nil {
+		return nil, err
+	}
+
+	u := *c.BaseURL
+	u.Path = path.Join(u.Path, rel.String(), sexOffenderSearchID)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.SetBasicAuth(c.APIKey, "")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, NewResponseError([]int{
+			resp.StatusCode,
+		}, resp)
+	}
+	defer func() {
+		if resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	getResp := &SexOffenderSearch{}
+	err = json.Unmarshal(b, getResp)
+	if err != nil {
+		return nil, err
+	}
+
+	return getResp, nil
+}
+
+const globalWatchListSearchPath = "/v1/global_watchlist_searches"
+
+// GetGlobalWatchListSearch ...
+func (r *Report) GetGlobalWatchListSearch(globalWatchlistSearchID string, c *Client) (*GlobalWatchListSearch, error) {
+
+	if globalWatchlistSearchID == "" {
+		return nil, errEmptyID
+	}
+
+	rel, err := url.Parse(globalWatchListSearchPath)
+	if err != nil {
+		return nil, err
+	}
+
+	u := *c.BaseURL
+	u.Path = path.Join(u.Path, rel.String(), globalWatchlistSearchID)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.SetBasicAuth(c.APIKey, "")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, NewResponseError([]int{
+			resp.StatusCode,
```

Comment:
@chandradeepak the status here should be the `expected` status code, so it should be `http.StatusOK`.